### ISSUE TITLE
Proper Way to Train LoRA for Voice Cloning Without Singer Encoder

### ISSUE THREAD

**Issue by pft-LoydWan** [ace-step/ACE-Step#259]
Hi, thank you for your excellent work on ACE-Step!

I noticed that the framework includes a singer encoder to condition vocals via cross-attention, but due to security and licensing concerns, we cannot use the pretrained singer encoder model.

As a workaround, I trained a LoRA on pure vocals (without background music) from Jay Chou’s song “青花瓷” for 1000 steps. However, the generated results appear overfitted — the output mimics the “青花瓷” melody, and the instrumental background disappears (Yeh, the result is expected I know).

My goal is to use the LoRA to clone the Jay Chou voice so that ACE-Step can generate new songs (e.g., singing different lyrics such as from “告白氣球”) while preserving the vocal timbre, but not the original melody or accompaniment.

Do you have any recommendations for:

How to prepare vocal datasets for LoRA training (how much vocal dataset, speech or singing vocal)?
How to avoid overfitting to melody and arrangement?
Whether LoRA is sufficient for voice cloning in ACE-Step, or if additional conditioning techniques are needed?
Thanks in advance!

the audio I trained on lora
https://github.com/user-attachments/assets/65d4996b-314e-41e2-95df-8bb6e27b36cf
and the generated song
https://github.com/user-attachments/assets/55f3a54d-d5c0-45a1-b676-8f6bcd0864c3

**Comment by woct0rdho** [comment]
I guess you can try to take the gradient of the training loss w.r.t. the speaker embedding vector, which will find the speaker embedding vector closest to your specific voice. I haven't tried how generalizable this method can be

**Comment by pft-LoydWan** [comment]
Oh, this sounds like a potential solution — I’ll give it a try, after the RVC method.

Currently, I’ve been doing some ablation studies on the three loss functions mentioned in ACE-Step (denoising, mHuBERT, and MERT) to train the lora, but the results actually got worse, likely because these losses are not designed to preserve speaker identity.

The diffusion denoising loss mainly aims to reconstruct the original spectrum, not voice timbre.
mHuBERT focuses more on speech representation, not speaker-specific vocal characteristics.
MERT is primarily designed for capturing musical structure, rather than vocal identit

**Comment by anujsinha72094** [comment]
@MJ54pwnz does the quality sound good ? what lora weight you set to get good results?
